### PR TITLE
SPLASH-32: Search Screen UI and Logic

### PR DESCRIPTION
**Pull request by Pepegaga**

### CODE REVIEW

File: src/app/search/sagas.ts [hunk 1]
```diff
@@ -6,8 +6,12 @@ import { client } from '../../graphql/api';
 import { SEARCH_ALL } from './actions';
 
 export function* searchSaga(action: ReturnType<typeof SEARCH_ALL.TRIGGER>): SagaIterator {
-    const result = yield call(client.search, action.payload);
-    yield put(SEARCH_ALL.STARTED({ text: action.payload, result: result }));
+    try {
+        const result = yield call(client.search, action.payload);
+        yield put(SEARCH_ALL.STARTED({ text: action.payload, result: result }));
```
Comment (TheLonelyAstronaut) [inline src/app/search/sagas.ts]: We should set isFetching with .STARTED() and set fetched data with .COMPLETED()

File: src/app/search/sagas.ts [hunk 2]
```diff
@@ -6,8 +6,12 @@ import { client } from '../../graphql/api';
...
+    } catch (e) {
+        console.log(e);
```
Comment (TheLonelyAstronaut) [inline src/app/search/sagas.ts]: Also implement .COMPLETED.failed() here

File: src/app/search/screens/search-screen.component.tsx [hunk 3]
```diff
@@ -1,12 +1,91 @@
...
+`;
 
 export const SearchScreenComponent: React.FC = () => {
+    const theme = useTheme();
+    const [search, setSearch] = useState('');
+    const dispatch = useDispatch();
+    const results = useSelector(getSearchResults);
+    const isFetching = useSelector(getIsFetching);
+
+    const renderItem: ListRenderItem<SearchResult> = ({ item }) => {
```
Comment (TheLonelyAstronaut) [inline src/app/search/screens/search-screen.component.tsx]: useCallback

File: src/app/search/screens/search-screen.component.tsx [hunk 4]
```diff
@@ -1,12 +1,91 @@
...
+        console.log(item);
+        if (item.type === SearchResultType.ARTIST) {
```
Comment (TheLonelyAstronaut) [inline src/app/search/screens/search-screen.component.tsx]: Two same returns from two if statements

File: src/app/ui/search-artis.component.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+`;
+export const ArtistName = styled(RegularText)`
+    color: ${(props) => props.theme.colors.secondary};
+`;
+export const ArtistText = styled(RegularText)`
+    color: ${(props) => props.theme.colors.inputBackground};
+    font-size: ${(props) => props.theme.fontSize.extraSmall};
+`;
+
+export const ArtistImage = styled.Image`
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-artis.component.tsx]: Use Image from /ui directory

File: src/app/ui/search-artis.component.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    height: ${(props) => props.theme.searchItem.searchImage}px;
+    border-radius: 100px;
+    margin-left: ${(props) => props.theme.spacer * 5};
+`;
+
+export type Props = {
+    title: string;
+    artist: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    image: any;
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-artis.component.tsx]: ))))))))))))))) We will change to string later

File: src/app/ui/search-artis.component.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-artis.component.tsx]: check filename)

File: src/app/ui/search-artis.component.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import styled from 'styled-components/native';
+
+import { RegularText } from './text.component';
+import { DEVICE_SIZE } from './themes/themes';
+
+export const Container = styled.TouchableOpacity`
+    width: ${DEVICE_SIZE.width};
+    height: ${(props) => props.theme.searchItem.height}px;
+    background-color: black;
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-artis.component.tsx]: styled components

File: src/app/ui/search-artis.component.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    margin-bottom: ${(props) => props.theme.spacer * 2};
+    margin-top: ${(props) => props.theme.spacer * 2};
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-artis.component.tsx]: Don't use margins here, use ListSeparatorComponent (FlatList prop)

File: src/app/ui/search-track.component.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+`;
+export const TrackName = styled(RegularText)`
+    color: ${(props) => props.theme.colors.secondary};
+`;
+export const ArtistText = styled(RegularText)`
+    color: ${(props) => props.theme.colors.inputBackground};
+    font-size: ${(props) => props.theme.fontSize.extraSmall};
+`;
+
+export const TrackImage = styled.Image`
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-track.component.tsx]: Image from /ui

File: src/app/ui/search-track.component.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (TheLonelyAstronaut) [inline src/app/ui/search-track.component.tsx]: Two same components? WTF bro, remove search-track and search-artist, add search-result.component.tsx

File: src/graphql/api.ts [hunk 12]
```diff
@@ -79,9 +81,39 @@ export class GraphQLAPI {
     // };
 
     search = async (name: string): Promise<SearchResult[]> => {
-        const result = tracks.filter((track) => track.title === name);
+        const artists1 = artists.filter((artist) => artist.name === name);
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: All search queries should be a substrings of results, not exactly result (For example, if you type 'Trav' server should response with Travis Scott, now it responses with empty array result)

File: src/types/music.ts [hunk 13]
```diff
@@ -18,3 +18,11 @@ export type Playlist = {
     name: string;
     id: number;
 };
+
+export type Artist = {
```
Comment (TheLonelyAstronaut) [inline src/types/music.ts]: albums should be an Album[] type, not Track[], Album will contain tracks, cover and other data. Also you forget about linkedArtists: Artist[] field, that should be optional. Check Artist Screen UI ticket to create correct Artist type

**Review by TheLonelyAstronaut — COMMENTED** [review]
Also you should display 'Nothing was found' in screen when server responses with empty array, you shouldn't send an error from server if nothing was found (just send an empty array, its ok).

File: src/types/music.ts (hunk 13 above)
Comment (Pepegaga) [reply src/types/music.ts]: It fixed in other PR

**Pull request merged**